### PR TITLE
doc(RECIPE): Add Request ID Logging recipe

### PR DESCRIPTION
**Pull request by rmyers**
Closes: #1500

* Adds a new section under User Guide for Recipes

Screen Shot 2019-05-06 at 11 26 20 AM

### CODE REVIEW

File: docs/user/recipes/request-id.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    from context import ctx
+
+    def create_widget_object(name: str) -> Any:
+        request_id = f'request_id={ctx.request_id}'
+        logging.debug(f'{request_id} going to create widget: {name}')
+        try:
+            # create the widget
+        except:
+            logging.exception(f'{request_id} something went wrong')
```
Comment (vytas7) [inline docs/user/recipes/request-id.rst]: We are still supporting Python 3.5, and IIRC format strings were introduced in Python 3.6.

File: docs/user/recipes/request-id.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    from uuid import uuid4
+
+    class RequestIDMiddleware:
+        def process_request(self, req, resp):
+            request_id = str(uuid4())
+            # Using Falcon 2.0 syntax
+            req.context.request_id = request_id
+
+            # Or if your logger has built-in support for contexts
+            req.context.log = structlog.get_logger(request_id=request_id)
```
Comment (vytas7) [inline docs/user/recipes/request-id.rst]: We should probably illustrate importing `structlog` as well in the same fashion as we do for `uuid`.

File: docs/user/recipes/request-id.rst (hunk 1 above)
Comment (rmyers) [reply docs/user/recipes/request-id.rst]: doh, ok, I'll fix that.

**Comment by rmyers** [comment]
@nZac @vytas7 I think this is ready now, had to fix a small typo. Please review again. Thanks!

**Pull request merged**